scripts/git_changes_utils_test.py

```diff
@@ -0,0 +1,567 @@
+            'that have the url https://github.com/oppia/oppia.git except for '
+            'the main \'upstream\' remote.\n' in self.print_arr)
+
+    def test_git_diff_name_status_without_error(self) -> None:
+        def mock_start_subprocess_for_result(
+            unused_cmd_tokens: List[str]
+        ) -> Tuple[bytes, None]:
+            return (b'M\tfile1\nA\tfile2', None)
+        subprocess_swap = self.swap(
+            common, 'start_subprocess_for_result',
+            mock_start_subprocess_for_result)
+
+        with subprocess_swap:
+            self.assertEqual(
+                git_changes_utils.git_diff_name_status(
+                    'left', 'right', diff_filter='filter'),
+                [
+                    git_changes_utils.FileDiff(status=b'M', name=b'file1'),
+                    git_changes_utils.FileDiff(status=b'A', name=b'file2')
+                ]
+            )
+
+    def test_git_diff_name_status_with_error(self) -> None:
+        def mock_start_subprocess_for_result(
+            unused_cmd_tokens: List[str]
+        ) -> Tuple[str, str]:
+            return ('M\tfile1\nA\tfile2', 'test_oppia_error')
+        subprocess_swap = self.swap(
+            common, 'start_subprocess_for_result',
+            mock_start_subprocess_for_result)
+
+        with subprocess_swap, self.assertRaisesRegex(
+            ValueError, 'test_oppia_error'
+        ):
+            git_changes_utils.git_diff_name_status(
+                'left', 'right', diff_filter='filter')
+
+    def test_git_diff_name_status_with_no_left_and_right(self) -> None:
+        def mock_start_subprocess_for_result(
+            cmd_tokens: List[str]
+        ) -> Tuple[bytes, None]:
+            if cmd_tokens == ['git', 'diff', '--name-status']:
+                return (b'M\tfile1\nA\tfile2', None)
+            return (b'', None)
+        subprocess_swap = self.swap(
+            common, 'start_subprocess_for_result',
+            mock_start_subprocess_for_result)
+
+        with subprocess_swap:
+            self.assertEqual(
+                git_changes_utils.git_diff_name_status(),
+                [
+                    git_changes_utils.FileDiff(status=b'M', name=b'file1'),
+                    git_changes_utils.FileDiff(status=b'A', name=b'file2')
+                ]
+            )
+
+    def test_git_diff_name_status_with_empty_left_should_error(self) -> None:
+        with self.assertRaisesRegex(
+            ValueError,
+            'Error: left should not be an empty string.'
+        ):
+            git_changes_utils.git_diff_name_status(left='')
+
+    def test_git_diff_name_status_with_empty_right_should_error(self) -> None:
+        with self.assertRaisesRegex(
+            ValueError,
+            'Error: right should not be an empty string.'
+        ):
+            git_changes_utils.git_diff_name_status(right='')
+
+    def test_git_diff_name_status_with_empty_diff_filter_should_error(self) -> None: # pylint: disable=line-too-long
```

Comment:
```suggestion
    def test_git_diff_name_status_with_empty_filter_should_error(self) -> None: # pylint: disable=line-too-long
```

To get the line length below 80 characters

Reply:
Done.